### PR TITLE
feat: add onboarding with oldmarr import

### PR DESCRIPTION
**Pull request by Meierschlumpf**
<br/>
<div align="center">
  
  <h3>Homarr</h3>
</div>

**Thank you for your contribution. Please ensure that your pull request meets the following pull request:**

- [x] Builds without warnings or errors (``pnpm buid``, autofix with ``pnpm format:fix``)
- [x] Pull request targets ``dev`` branch
- [x] Commits follow the [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] No shorthand variable names are used (eg. ``x``, ``y``, ``i`` or any abbrevation)

**Start screen**
start

**Import dropzone**
import-dropzone

**Import options and summary**
import-options-and-summary

**Token modal for secrets & user credentials**
token-modal

**Init user**
init-user

**Init external group**
init-group

**Server settings**
server-settings

**Finish**
finish

### CODE REVIEW

File: apps/nextjs/src/app/[locale]/init/_steps/finish/init-finish.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import { createDocumentationLink } from "@homarr/definitions";
+import { getScopedI18n } from "@homarr/translation/server";
+import type { TablerIcon } from "@homarr/ui";
+
+export const InitFinish = async () => {
+  const firstBoard = await db.query.boards.findFirst({ columns: { name: true } });
+  const tFinish = await getScopedI18n("init.step.finish");
+
+  return (
+    <Card w={64 * 6} maw="90vw" withBorder>
```
Comment (Meierschlumpf) [inline apps/nextjs/src/app/[locale]/init/_steps/finish/init-finish.tsx:18]: Use shared card. Also adjust colors

File: apps/nextjs/src/app/[locale]/init/_steps/group/init-group.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        <Stack>
+          <TextInput
+            label={t("init.step.group.form.name.label")}
+            description={t("init.step.group.form.name.description")}
+            withAsterisk
+            {...form.getInputProps("name")}
+          />
+          <Button type="submit" loading={form.submitting} rightSection={<IconArrowRight size={16} stroke={1.5} />}>
+            {t("common.action.continue")}
+          </Button>
```
Comment (Meierschlumpf) [inline apps/nextjs/src/app/[locale]/init/_steps/group/init-group.tsx:47]: Add previous buttons

File: packages/api/src/router/onboard/onboard-queries.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  start: {
+    import: "preferred" as const,
+    user: () => env.AUTH_PROVIDERS.includes("credentials"),
+    group: () => env.AUTH_PROVIDERS.includes("ldap") || env.AUTH_PROVIDERS.includes("oidc"),
+    settings: true,
+  },
+  import: {
+    // eslint-disable-next-line no-restricted-syntax
+    user: async (db: Database) => {
+      if (!env.AUTH_PROVIDERS.includes("credentials")) return false;
```
Comment (Meierschlumpf) [inline packages/api/src/router/onboard/onboard-queries.ts:57]: Use `isProviderEnabled` instead

File: packages/db/migrations/seed.ts [hunk 4]
```diff
@@ -31,6 +32,20 @@ const seedEveryoneGroupAsync = async (db: Database) => {
...
+
+  if (existing) {
+    console.log("Skipping seeding of onboarding as it already exists");
+    return;
+  }
+
+  await db.insert(onboarding).values({
+    id: createId(),
+    step: "start",
+  });
```
Comment (Meierschlumpf) [inline packages/db/migrations/seed.ts:46]: Add log for successful insert

File: packages/old-import/src/import/import-initial-oldmarr.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  db.transaction((transaction) => {
+    boardInsertCollection.insertAll(transaction);
+    userInsertCollection.insertAll(transaction);
+    integrationInsertCollection.insertAll(transaction);
+  });
+
+  logger.info(`Import successful (in ${stopwatch.getElapsedInHumanWords()})`);
+};
+
+// TODO: Add note about screen size feature no longer exists
```
Comment (Meierschlumpf) [inline packages/old-import/src/import/import-initial-oldmarr.ts:47]: Pending ToDo

File: packages/old-import/src/import/import-single-oldmarr.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import type { OldmarrImportConfiguration } from "../settings";
+import { createBoardInsertCollection } from "./collections/board-collection";
+
+export const importSingleOldmarrConfig = (
+  db: Database,
+  config: OldmarrConfig,
+  settings: OldmarrImportConfiguration,
+) => {
+  const { preparedApps, preparedBoards } = prepareSingleImport(config, settings);
+  // TODO: We should only insert apps that are not already in the database, so we need to check for that
```
Comment (Meierschlumpf) [inline packages/old-import/src/import/import-single-oldmarr.ts:14]: Pending ToDo

File: packages/old-import/src/import/input.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+const boardSelectionMapSchema = z.map(
+  z.string(),
+  z.object({
+    sm: z.boolean().nullable(),
+    md: z.boolean().nullable(),
+    lg: z.boolean().nullable(),
+  }),
+);
+
+export const importInitialOldmarrInputSchema = zfd.formData({
```
Comment (Meierschlumpf) [inline packages/old-import/src/import/input.ts:16]: Remove those from validation package

File: packages/old-import/src/mappers/map-integration.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  tdarr: null,
+  transmission: "transmission",
+  plex: "plex",
+};
+
+export const mapAndDecryptIntegrations = (
+  preparedIntegrations: PreparedIntegration[],
+  encryptionToken: string | null,
+) => {
+  // TODO: Remove this when the encryption is implemented
```
Comment (Meierschlumpf) [inline packages/old-import/src/mappers/map-integration.ts:43]: Remove

File: packages/old-import/src/mappers/map-user.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
+import { decryptSecretWithKey } from "@homarr/common/server";
+import type { InferInsertModel } from "@homarr/db";
+import { createId } from "@homarr/db";
+import type { users } from "@homarr/db/schema/sqlite";
+
+import type { OldmarrImportUser } from "../user-schema";
+
+export const mapAndDecryptUsers = (importUsers: OldmarrImportUser[], encryptionToken: string | null) => {
+  // TODO: Remove this when the encryption is implemented
```
Comment (Meierschlumpf) [inline packages/old-import/src/mappers/map-user.ts:9]: Remove

File: packages/old-import/src/prepare/prepate-sections.ts
Comment (Meierschlumpf) [inline packages/old-import/src/prepare/prepate-sections.ts:1]: Fix typo in filename

File: packages/old-import/src/settings.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import { z } from "zod";
+
+import { boardSizes } from "@homarr/old-schema";
+import { validation } from "@homarr/validation";
+
+import { zodEnumFromArray } from "../../validation/src/enums";
```
Comment (Meierschlumpf) [inline packages/old-import/src/settings.ts:6]: Wrong import

File: packages/old-schema/src/tile.ts [hunk 11]
```diff
@@ -1,5 +1,7 @@
 import { z } from "zod";
 
+import { objectKeys } from "../../common/src";
```
Comment (Meierschlumpf) [inline packages/old-schema/src/tile.ts:3]: Wrong import

File: packages/db/migrations/seed.ts [hunk 12]
```diff
@@ -31,6 +32,21 @@ const seedEveryoneGroupAsync = async (db: Database) => {
...
+  console.log("Created onboarding step through seed");
```
Comment (manuel-rw) [inline packages/db/migrations/seed.ts:47]: Shouldn't we use the logger here?

File: packages/old-import/src/components/shared/apps-section.tsx [hunk 13]
```diff
@@ -0,0 +1,23 @@
+import { Fieldset, Switch } from "@mantine/core";
+
+import type { CheckboxProps } from "@homarr/form/types";
+import { useScopedI18n } from "@homarr/translation/client";
+
+interface OldmarrImportAppsSettingsProps {
+  onlyImportApps: CheckboxProps;
+  bg?: string;
```
Comment (manuel-rw) [inline packages/old-import/src/components/shared/apps-section.tsx:8]: NIT, rename to background?

File: packages/old-import/src/prepare/prepare-boards.ts [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+import type { BoardSize } from "@homarr/old-schema";
+
+import type { ValidAnalyseConfig } from "../analyse/types";
+import type { BoardSelectionMap } from "../components/initial/board-selection-card";
+
+const boardSizeSuffix: Record<BoardSize, string> = {
+  lg: "large",
+  md: "medium",
+  sm: "small",
+};
```
Comment (manuel-rw) [inline packages/old-import/src/prepare/prepare-boards.ts:11]: Nice 👍

**Review by manuel-rw — APPROVED** [review]

File: packages/db/migrations/seed.ts (hunk 12 above)
Comment (Meierschlumpf) [reply packages/db/migrations/seed.ts:47]: Probably not really possible except if I add additional logic for it. Not worth it IMO

**Review by manuel-rw — APPROVED** [review]

**Pull request merged**